homeassistant/components/melnor/sensor.py

```diff
@@ -60,6 +97,17 @@ class MelnorSensorEntityDescription(
     ),
 ]
 
+ZONE_ENTITY_DESCRIPTIONS: list[MelnorZoneSensorEntityDescription] = [
+    MelnorZoneSensorEntityDescription(
```

Comment:
We want to avoid seconds that write state frequently 

Usually we use `SensorDeviceClass.TIMESTAMP` and set the time when it will be done instead so we spam the state machine with lots of updates

Reply:
For this use case I'm going to have to strongly argue against using `TIMESTAMP`. Resolving the remaining time to seconds is crazy, I agree. I've added a little bit of extra math to snap to the current number of minutes remaining instead. This removes the excess noise from the state machine while retaining a useful history view for end users.

The history view for a 60 minute runtime with `Timestamp`
<img width="595" alt="image" src="https://user-images.githubusercontent.com/577219/190936965-a3986d90-5b43-43ac-9953-5d3ac5cc8393.png">

View of `DURATION`with a resolution of minutes, rather than seconds.
<img width="594" alt="image" src="https://user-images.githubusercontent.com/577219/190937319-944b77fb-7a3d-4d89-b67e-eaaf288d4333.png">

Reply:
I'm afraid there isn't flexibility on this one.  I can't merge it as-is.

Here is an example of one that was recently converted.
https://github.com/home-assistant/core/blob/9655f30146aab006b76755283267719b658d21a6/homeassistant/components/rainmachine/sensor.py#L183

Reply:
Frustrating that the user experience on the frontend is so poor for this sensor class, though I can respect that a friendlier representation is non-trivial given how broad a timestamp can be. I'll add this to my list of core stuff I'd like to see improved as I'd like to swing that way once I wrap up this current work.

Reply:
If you drop the UOM it will look like this on the frontend

<img width="537" alt="Screen Shot 2022-09-19 at 6 43 39 AM" src="https://user-images.githubusercontent.com/663432/191009841-7000e38e-9375-44fd-8e15-095112121bdb.png">

Reply:
For clarity, if you delete the old states from the database that had the UOM it should change how its represented. The frontend is showing a graph since the first state in the time window had a UOM of seconds.

Reply:
Might be best to start with a frontend PR as a GHI will likely be a candidate for stale bot unless someone else is passionate about it.

Reply:
Good advice, thanks!